### ISSUE TITLE
era5land option in get_era5.py

### ISSUE THREAD

**Issue by StephanThober** [mcuntz/jams_python#6]
Hi Matthias,

I included the option for era5land in get_era5.py. This option is, however, very unstable. I get frequent error messages from CDS making this option almost not working. I am wondering, how to proceed. There are three option:

1.) leave it as it is.
2.) issue a message to the user that this option is experimental and might not work.
3.) revert to a previous version that did not contain this option.

I am looking forward to your opinion.

Best,
Stephan

**Comment by mcuntz** [comment]
Hey Stephan,

I have worked on the script recently, cleaning some bugs and polishing the era5-land option. I hope that it is more robust now.

Era5-land takes a lot of time to extract. So one possibility is that your computer goes to sleep, which might break the link. But the request should continue and you should be able to download the file later from the website.

Another issue is that (at the time I wrote the original script), CDS only allowed downloading 100,000 item at a time. Items are time steps and variables, but a map is one item. So 10 variables, 24 hours a day per year are: 10*24*365 = 87600. That's why the script downloads one file per year. Be sure that you stay below 10 variables at once.

I just downloaded era5-land for two years at our flux station. It took more than one hour to process. I already waited 18 hours for 10 variables from 1981-2019 at one single point. But all jobs finished.
I have no experience with larger grids. I downloaded era5 but not era5-land.

Could you give it another try please and look if the updated script is more robust.

Kind regards
Matthias

**Comment by mcuntz** [comment]
Hey Stephan,

did exactly the same request but using era5 instead of era5-land:
python ./get_era5.py -r ERA5 -v 2m_temperature,2m_dewpoint_temperature -a 48.6742166667,7.06461666667 -y 2018,2019 -p era

This took 15 minutes. Much more reasonable. Perhaps we could find out why era5-land takes some much longer. Might be an interpolation thing. We could try to get era5-land on the natural resolution and do the interpolation ourselves, if needed.

Matthias

**Comment by mcuntz** [comment]
Hey @StephanThober

are you still using it? I downloaded ERA5-Land with the script and it works fine. It is only that the file preparation takes a lot of time and the terminal stalls. So I basically asked for 8 years of data over Western Europe; the script started 8 requests; and then I just hit CTRL-C, leaving the waiting prompt. But file preparation continues to happen on the ECMWF servers and a few hours later, I download the files with a web browser. Only then I restarted the next 8 years.

I realised if I asked for too many years at once, some were already inaccessible when I tried to download them.

Kind regards,
Matthias
